### PR TITLE
Fix client mode and clean-up

### PR DESCRIPTION
**Pull request by imobachgs**
This PR tries to address some issues:

* Fix client mode (at least for Salt provisioner).
* The module configuration is not written to disk anymore because all happens now during the 1st stage. Less code to maintain and break :)
* Merge CFA classes to handle minions configuration into a single class.

Anyway, it is still a work in progress.

### CODE REVIEW

File: src/lib/configuration_management/cfa/minion.rb [hunk 1]
```diff
@@ -1,20 +1,16 @@
...
-        attributes(master: "master")
-
         # Configuration parser
-        # FIXME: At this time, we're using Augeas' cobblersettings lense,
-        # as long as YAML is not supported. We should use another parser.
-        PARSER = ::CFA::AugeasParser.new("cobblersettings.lns")
+        PARSER = Yast::ConfigurationManagement::CFA::YAMLParser.new
         # Path to configuration file
-        PATH = "/etc/salt/minion".freeze
+        PATH = "/etc/salt/minion.d/yast-configuration-management.conf".freeze
```
Comment (schubi2) [inline src/lib/configuration_management/cfa/minion.rb]: Very nice to concentrate all YaST specific stuff to one own configuration file.

File: package/yast2-configuration-management.changes [hunk 2]
```diff
@@ -1,4 +1,14 @@
 -------------------------------------------------------------------
+Tue May  9 14:07:13 UTC 2017 - igonzalezsosa@suse.com
+
+- Fix masterless mode (bsc#1038290)
+- Fix Puppet package installation
+- Handle libzypp's lock to allow installation of packages
+  through Salt/Puppet
+- Do not write configuration to disk as it is not needed anymore
```
Comment (schubi2) [inline package/yast2-configuration-management.changes]: I have also thought about it some time ago. Do we have all information for a "yast clone_system" and the coming up UI in
/etc/salt/minion.d/yast-configuration-management.conf or would some additional information needed for it ?

**Comment by schubi2** [comment]
Great improvements ! Just a little thought about the stored settings but feel free to ignore it :-) LGTM

File: package/yast2-configuration-management.changes (hunk 2 above)
Comment (imobachgs) [inline package/yast2-configuration-management.changes]: No, we do not have all the details. The configuration that was written, contained some specific details to allow running the provisioner after YaST finished.

IMHO, the `yast clone_system` and UI can be slightly different to the original intention (some information like the path where states/recipes were downloaded are not needed at all). So probably we'll have to bring back part of that code, but I would do it when we implement such features.

**Pull request merged**